stixcore/io/fits/processors.py

```diff
@@ -499,7 +505,7 @@ def write_fits(self, product, path=None, *, version=1):
                                   for version, range in prod.idb_versions.items()],
                                   names=["version", "obt_start", "obt_end"])
 
-            primary_header = self.generate_primary_header(filename, prod, version=version)
+            primary_header = self.generate_primary_header(filename_header, prod, version=version)
```

Comment:
So the name of the file and the filename header will be different?

Reply:
s.a.:

Reply:
updated so the the FILENAME keyword also has the same value as the filename. will be updated later again